openhands/tools/__init__.py

```diff
@@ -43,10 +24,82 @@
     "BrowserToolSet",
 ]
 
-from importlib.metadata import PackageNotFoundError, version
-
-
 try:
     __version__ = version("openhands-tools")
 except PackageNotFoundError:
     __version__ = "0.0.0"  # fallback for editable/unbuilt environments
+
+
+_mapping = {
+    # execute_bash
+    "execute_bash_tool": ("openhands.tools.execute_bash", "execute_bash_tool"),
+    "ExecuteBashAction": ("openhands.tools.execute_bash", "ExecuteBashAction"),
+    "ExecuteBashObservation": (
+        "openhands.tools.execute_bash",
+        "ExecuteBashObservation",
+    ),
+    "BashExecutor": ("openhands.tools.execute_bash", "BashExecutor"),
+    "BashTool": ("openhands.tools.execute_bash", "BashTool"),
+    # str_replace_editor
+    "str_replace_editor_tool": (
+        "openhands.tools.str_replace_editor",
+        "str_replace_editor_tool",
+    ),
+    "StrReplaceEditorAction": (
+        "openhands.tools.str_replace_editor",
+        "StrReplaceEditorAction",
+    ),
+    "StrReplaceEditorObservation": (
+        "openhands.tools.str_replace_editor",
+        "StrReplaceEditorObservation",
+    ),
+    "FileEditorExecutor": ("openhands.tools.str_replace_editor", "FileEditorExecutor"),
+    "FileEditorTool": ("openhands.tools.str_replace_editor", "FileEditorTool"),
+    # task_tracker
+    "task_tracker_tool": ("openhands.tools.task_tracker", "task_tracker_tool"),
+    "TaskTrackerAction": ("openhands.tools.task_tracker", "TaskTrackerAction"),
+    "TaskTrackerObservation": (
+        "openhands.tools.task_tracker",
+        "TaskTrackerObservation",
+    ),
+    "TaskTrackerExecutor": ("openhands.tools.task_tracker", "TaskTrackerExecutor"),
+    "TaskTrackerTool": ("openhands.tools.task_tracker", "TaskTrackerTool"),
+    # browser_use (heavy; only loads if you actually touch it)
+    "BrowserToolSet": ("openhands.tools.browser_use", "BrowserToolSet"),
+}
+
+
+def __getattr__(name: str):
```

Comment:
This magic tools reasonable but might make sure there are unit tests that exercise it. Hopefully they already do.